packages/gatsby/src/redux/index.js

```diff
@@ -122,14 +122,10 @@ exports.getNodeAndSavePathDependency = (id, path) => {
 exports.getRootNodeId = node => rootNodeMap.get(node)
 
 const addParentToSubObjects = (data, parentId) => {
-  _.each(data, (v, k) => {
-    if (_.isArray(v) && _.isObject(v[0])) {
-      _.each(v, o => addParentToSubObjects(o, parentId))
-    } else if (_.isObject(v)) {
-      addParentToSubObjects(v, parentId)
-    }
-  })
-  rootNodeMap.set(data, parentId)
+  if (_.isObject(data)) {
```

Comment:
I never expect this method from lodash to work like this.  it might be clearer (slightly) to do  `data != null && typeof data == 'object'`, or it might just be me that thinks its behavior is unexpected

Reply:
for correctness tho this really shouldn't work for Date, Number(0), Regex etc, and probably not functions either

Reply:
`typeof ['some', 'array', 'elements']` will be `'object'`. I was actually suprised that lodash will return true for arrays in `_.isObject` but we learn something new everyday it seems.

Condition you wrote is actually pretty much `_.isObjectLike` which maybe is something i should use - but this check shouldn't be there - im not really validating correctness of data there - this should be be done earlier in `createNode` function (before function that links subnodes to root node), right?

Reply:
from what i can see from the docs `_.isObject` behaves like typeof object but fails for `null`. In anycase it's not about data correctness here it's about whether the function should recurse down further, which is something we should only do for "true" objects and arrays, e.g things that correspond to GQL object and List types